Tools/ros2/ardupilot_sitl/CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+
+# TODO(srmainwaring): add cache variables for vehicle type, debug etc. 
+# set(ARDUPILOT_VEHICLE "copter" CACHE STRING "Vehicle type")
+
+# NOTE: look for `waf` and set source and build directories to top level.
+# ${PROJECT_SOURCE_DIR} = ./Tools/ros2/ardupilot
+# 
+cmake_path(SET ARDUPILOT_ROOT NORMALIZE ${PROJECT_SOURCE_DIR}/../../..)
+# message("ARDUPILOT_ROOT: ${ARDUPILOT_ROOT}")
+
+find_program(WAF_COMMAND waf HINTS ${ARDUPILOT_ROOT})
+message(STATUS "WAF_COMMAND: ${WAF_COMMAND}")
+
+add_custom_target(ardupilot_configure ALL
+  ${WAF_COMMAND} configure --board sitl --enable-dds --debug
+  WORKING_DIRECTORY ${ARDUPILOT_ROOT}
+)
+
+add_custom_target(ardupilot_build ALL
+  ${WAF_COMMAND} build --enable-dds --debug
+  WORKING_DIRECTORY ${ARDUPILOT_ROOT}
+)
+add_dependencies(ardupilot_build ardupilot_configure)
+# ament_target_dependencies((ardupilot_build DEPENDENCIES)
+
+# ament_export_targets(ardupilot_buildTargets HAS_LIBRARY_TARGET)
+# ament_export_dependencies()
+
+# --------------------------------------------------------------------------- #
+#  install
+
+# install targets (ament)
+# install(
+#   TARGETS # add targets
+#   EXPORT  # add build targets e.g. ardupilot_buildTargets
+#   LIBRARY DESTINATION lib
+#   ARCHIVE DESTINATION lib
+#   RUNTIME DESTINATION bin
+#   INCLUDES DESTINATION include
+# )
+
+# install executables
+install(PROGRAMS
+  ${ARDUPILOT_ROOT}/build/sitl/bin/antennatracker
+  ${ARDUPILOT_ROOT}/build/sitl/bin/arducopter
+  ${ARDUPILOT_ROOT}/build/sitl/bin/arducopter-heli
+  ${ARDUPILOT_ROOT}/build/sitl/bin/arduplane
+  ${ARDUPILOT_ROOT}/build/sitl/bin/ardurover
+  ${ARDUPILOT_ROOT}/build/sitl/bin/ardusub
+  ${ARDUPILOT_ROOT}/build/sitl/bin/blimp
+  DESTINATION bin/
+)
+
+# install libs
+install(DIRECTORY
+  ${ARDUPILOT_ROOT}/build/sitl/lib
+  DESTINATION ./
+)
+
+# install launch files
+install(DIRECTORY
+  launch
+  DESTINATION share/${PROJECT_NAME}/
+)
+
+# install DDS profile
+install(FILES
+  ${ARDUPILOT_ROOT}/libraries/AP_DDS/dds_xrce_profile.xml
+  DESTINATION share/${PROJECT_NAME}/config
+)
+
+# install additional default params
+install(DIRECTORY
+  config/default_params
+  DESTINATION share/${PROJECT_NAME}/config
+)
+
+# install autotest default params
+install(DIRECTORY
+  ${ARDUPILOT_ROOT}/Tools/autotest/default_params
+  DESTINATION share/${PROJECT_NAME}/config/
+)
+
+# install Python package
+ament_python_install_package(${PROJECT_NAME})
+
+# --------------------------------------------------------------------------- #
+# build tests
+
+if(BUILD_TESTING)
+  # add python tests
+  find_package(ament_cmake_pytest REQUIRED)
+  set(_pytest_tests
+    tests/test_a.py
+  )
+  foreach(_test_path ${_pytest_tests})
+  get_filename_component(_test_name ${_test_path} NAME_WE)
+  ament_add_pytest_test(${_test_name} ${_test_path}
+    APPEND_ENV PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}
+    TIMEOUT 60
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  )
+  endforeach()
+  
```

Comment:
Can you add in ament_lint_common, or something similar to enfore style/syntax in XML, python, and cmake code?

Reply:
I've enabled `ament_lint_auto` in `ardupilot_sitl` to see how it looks. That package is now compliant with all linters **and** formatted with `black`. There is a way to override the default ament flake8 config. The overrides are in the `.flake8` config file. There are only two changes needed - line length and double quotes. There may be more as we add code, but it seems to be relatively pain free.